app/assets/javascripts/discourse/app/components/modal/avatar-selector.hbs

```diff
@@ -0,0 +1,132 @@
+<DModal
+  @bodyClass="avatar-selector"
+  @closeModal={{@closeModal}}
+  @title={{i18n "user.change_avatar.title"}}
+  class="avatar-selector-modal"
+>
+  <:body>
+    {{#if this.showSelectableAvatars}}
+      <div class="selectable-avatars">
+        {{#each this.selectableAvatars as |avatar|}}
+          <a
+            href
+            class="selectable-avatar"
+            {{on "click" (fn this.selectAvatar avatar)}}
+          >
+            {{bound-avatar-template avatar "huge"}}
+          </a>
+        {{/each}}
+      </div>
+      {{#if this.showAvatarUploader}}
+        <h4>{{html-safe (i18n "user.change_avatar.use_custom")}}</h4>
+      {{/if}}
+    {{/if}}
+    {{#if this.showAvatarUploader}}
+      {{#if this.user.use_logo_small_as_avatar}}
+        <div class="avatar-choice">
+          <RadioButton
+            @id="logo-small"
+            @name="logo"
+            @value="logo"
+            @selection={{this.selected}}
+            @onChange={{this.onSelectedChanged}}
+          />
+          <label class="radio" for="logo-small">{{bound-avatar-template
+              this.siteSettings.site_logo_small_url
+              "large"
+            }}
+            {{html-safe (i18n "user.change_avatar.logo_small")}}</label>
+        </div>
+      {{/if}}
+      <div class="avatar-choice">
+        <RadioButton
+          @id="system-avatar"
+          @name="avatar"
+          @value="system"
+          @selection={{this.selected}}
+          @onChange={{this.onSelectedChanged}}
+        />
+        <label class="radio" for="system-avatar">{{bound-avatar-template
+            this.user.system_avatar_template
+            "large"
+          }}
+          {{html-safe (i18n "user.change_avatar.letter_based")}}</label>
+      </div>
+      {{#if this.allowAvatarUpload}}
+        <div class="avatar-choice">
+          <RadioButton
+            @id="gravatar"
+            @name="avatar"
+            @value="gravatar"
+            @selection={{this.selected}}
+            @onChange={{this.onSelectedChanged}}
+          />
+          <label class="radio" for="gravatar">{{bound-avatar-template
+              this.user.gravatar_avatar_template
+              "large"
+            }}
+            <span>{{html-safe
+                (i18n "user.change_avatar.gravatar" ...this.gravatar)
+              }}
+              {{this.user.email}}</span></label>
```

Comment:
This doesn't work. There is no spread operator in hbs (`...attributes` being a special case)

That's the reason tests failed 😛 

```suggestion
            <span>
              {{html-safe
                (i18n
                  "user.change_avatar.gravatar"
                  gravatarName=this.gravatar.gravatarName
                  gravatarBaseUrl=this.gravatar.gravatarBaseUrl
                  gravatarLoginUrl=this.gravatar.gravatarLoginUrl
                )
              }}
              {{this.user.email}}
            </span>
          </label>
```

Reply:
There is no destructuring in hbs. Depending on the number of dots before a reference you'll still pass in that whole object (as a positional argument)

I _think_ the i18n helper supports that? so it could be used just as `(i18n "user.change_avatar.gravatar" this.gravatar)`